test/geocel/GeantGdmlLoader.test.cc

```diff
@@ -0,0 +1,147 @@
+{
+    ScopedLogStorer scoped_log_local_{&celeritas::self_logger(),
+                                      LogLevel::warning};
+    auto* world = load_gdml(this->gdml_path("solids"));
+    static char const* const expected_local_log_levels[] = {"error"};
+    EXPECT_VEC_EQ(expected_local_log_levels, scoped_log_local_.levels());
+    scoped_log_local_ = {};
+
+    ASSERT_TRUE(world);
+
+    auto vols = find_geant_volumes({"box500", "trd3", "trd1"});
+
+    save_gdml(world, this->make_unique_filename(".gdml"));
+
+    static char const* const expected_log_messages[] = {
+        R"(Geant4 regions have not been set up: skipping export of energy cuts and regions)"};
+    EXPECT_VEC_EQ(expected_log_messages, scoped_log_.messages());
+    static char const* const expected_log_levels[] = {"warning"};
+    EXPECT_VEC_EQ(expected_log_levels, scoped_log_.levels());
+}
+
+using SimpleCmsTest = GeantGdmlLoaderTest;
+
+TEST_F(SimpleCmsTest, detectors)
+{
+    GeantGdmlLoader::Options opts;
+    opts.detectors = true;
+
+    GeantGdmlLoader load_gdml(opts);
+    auto loaded = load_gdml(this->gdml_path("simple-cms"));
+    EXPECT_TRUE(loaded.world);
+    EXPECT_EQ(1, loaded.detectors.count("si_tracker_sd"));
+    EXPECT_EQ(1, loaded.detectors.count("em_calorimeter_sd"));
+
+    EXPECT_TRUE(scoped_log_.empty()) << scoped_log_;
+}
+
+using CmsEeTest = GeantGdmlLoaderTest;
+
+TEST_F(CmsEeTest, ignore)
+{
+    GeantGdmlLoader::Options opts;
+    opts.detectors = true;
+    opts.pointers = PT::ignore;
+
+    GeantGdmlLoader load_gdml(opts);
+    auto loaded = load_gdml(this->gdml_path("cms-eebackdee"));
+
+    EXPECT_EQ(2, loaded.detectors.count("ee_back_plate"));
+    EXPECT_EQ(2, loaded.detectors.count("ee_s_ring"));
+
+    // Reflected volume name is intact
+    SetLV found;
+    EXPECT_NO_THROW(found = find_geant_volumes({"EEBackQuad0x7f4a8f07c900",
+                                                "EEBackQuad0x7f4a8f07c900_"
+                                                "refl"}));
+    EXPECT_EQ(2, found.size());
+    EXPECT_TRUE(scoped_log_.empty()) << scoped_log_;
+}
+
+TEST_F(CmsEeTest, amputate)
+{
+    GeantGdmlLoader::Options opts;
+    opts.detectors = true;
+    opts.pointers = PT::amputate;
+
+    GeantGdmlLoader load_gdml(opts);
+    auto loaded = load_gdml(this->gdml_path("cms-eebackdee"));
+
+    EXPECT_EQ(2, loaded.detectors.count("ee_back_plate"));
+    EXPECT_EQ(2, loaded.detectors.count("ee_s_ring"));
+
+    // Reflected volume name is deleted by G4 GDML parser
+    SetLV found;
+    EXPECT_NO_THROW(found = find_geant_volumes({"EEBackQuad"}));
+    EXPECT_EQ(2, found.size());
+
+    static char const* const expected_log_messages[]
+        = {"Multiple Geant4 volumes are mapped to name 'EEBackQuad'"};
+    EXPECT_VEC_EQ(expected_log_messages, scoped_log_.messages());
+    static char const* const expected_log_levels[] = {"warning"};
+    EXPECT_VEC_EQ(expected_log_levels, scoped_log_.levels());
+}
+
+TEST_F(CmsEeTest, excise)
```

Comment:
Do we have to worry about round-tripping the GDML load/write?

Reply:
It seems the under-the-hood antics of the reflection factory mean that 'remove' and 'truncate' produce the same file (modulo pointer addresses) as the input. The `ignore` option results in two pointers in the output.

Reply:
Hmm, when I try to load this saved file I don't see any sensitive detectors... but they should be saved, right?

Reply:
If we attached the SDs inside the code, we would; Geant4's GDML loader doesn't actually do that (and for the unit test we don't either). If we did this as part of a celer-g4 run I think it would round trip correctly.